File: javascript/src/voice/voice-models.ts
/**
 * Default model identifiers for voice paths.
 *
 * Single source of truth so demos, adapters, and any stub bot don't drift.
 * Anything outside this module that needs a model name should import from
 * here.
 *
 * Mirror of `python/scenario/config/voice_models.py`. When OpenAI / Google /
 * ElevenLabs ship a new generation, update this file and re-test once.
 * Avoid sprinkling magic strings.
 */

/** OpenAI Realtime — bidirectional streaming audio API. */
export const OPENAI_REALTIME_MODEL = "gpt-realtime-mini";

/** OpenAI STT — speech-to-text default for the judge / transcript path. */
export const OPENAI_STT_MODEL = "gpt-4o-transcribe";

/** OpenAI TTS — text-to-speech endpoint default. */
export const OPENAI_TTS_MODEL = "gpt-4o-mini-tts";

/** Gemini Live — bidirectional audio (Google's realtime model). */
export const GEMINI_LIVE_MODEL = "gemini-2.5-flash-native-audio-latest";

/**
 * Composable voice agent LLM — text-only, drives the brain of the
 * {@link ComposableVoiceAgent} and {@link ElevenLabsVoiceAgent} stacks
 * (STT → LLM → TTS). Python parity: `COMPOSABLE_VOICE_LLM_MODEL` in
 * `python/scenario/config/voice_models.py`. The Python identifier is the
 * litellm-style string `"openai/gpt-5.4-mini"`; the TS branded preset
 * resolves this to `openai("gpt-5.4-mini")` via the `@ai-sdk/openai`
 * factory.
 */
export const COMPOSABLE_VOICE_LLM_MODEL = "gpt-5.4-mini";

/**
 * Default ElevenLabs TTS model. `eleven_v3` is the only model that honors
 * inline paralinguistic markers like `[shouting]`, `[laughs]`. The SDK's
 * default (`eleven_multilingual_v2`) reads them as text — see Python's
 * `tts.py:107` for the angry-customer regression that drove the hardcode.
 */
export const ELEVENLABS_TTS_MODEL = "eleven_v3";

/** ElevenLabs STT model — Python parity: `stt.py:84`. */
export const ELEVENLABS_STT_MODEL = "scribe_v1";

/**
 * Sarah — premade EL voice, free-tier accessible as of 2026-05. Other
 * premade voices (e.g. Rachel `21m00Tcm4TlvDq8ikWAM`) returned 402
 * paid_plan_required from the EL TTS API. Override via `ELEVENLABS_VOICE_ID`
 * env or the `voice` constructor argument.
 */
export const ELEVENLABS_DEFAULT_VOICE_ID = "EXAVITQu4vr4xnSDxMaL";

File: python/scenario/config/voice_models.py
"""
Default model identifiers for voice paths.

Single source of truth so demos, adapters, and the stub bot don't drift.
Anything outside `config/` that needs a model name should import from here.

Update strategy: when OpenAI / Google / ElevenLabs ship a new generation,
update this file and re-test once. Avoid sprinkling magic strings.
"""

# OpenAI Realtime — bidirectional streaming audio API.
# Used by OpenAIRealtimeAgentAdapter and the Pipecat-Twilio reference bot.
# `gpt-realtime-mini` is the cost-efficient GA model; `gpt-realtime-1.5`
# is the premium tier (~3× more expensive) for higher voice quality.
OPENAI_REALTIME_MODEL: str = "gpt-realtime-mini"

# OpenAI chat completions for the stub bot's brain (STT → chat → TTS
# pipeline). `gpt-5.4-nano` is the cheapest current-GA chat model;
# bot replies cap at 60 tokens so we don't need mini-tier capability.
OPENAI_BOT_LLM_MODEL: str = "gpt-5.4-nano"

# OpenAI TTS — text-to-speech endpoint. `gpt-4o-mini-tts` supersedes
# `tts-1`; same API surface, modern voices.
# Deliberate: gpt-4o-mini-tts is OpenAI's current-gen TTS model as of 2026-06
# (no gpt-5-tts exists on the public API). Revisit when OpenAI ships a
# gpt-5-family speech successor.
OPENAI_TTS_MODEL: str = "gpt-4o-mini-tts"

# OpenAI STT — speech-to-text. Used by voice adapters that need to
# transcribe audio (e.g. the Realtime adapter's input_audio_transcription
# config). Standard tier — accuracy here directly drives downstream
# judge correctness on non-multimodal paths.
# Deliberate: gpt-4o-transcribe is OpenAI's current-gen transcription model
# as of 2026-06 (no gpt-5-transcribe exists on the public API). Revisit when
# OpenAI ships a gpt-5-family transcription successor.
OPENAI_STT_MODEL: str = "gpt-4o-transcribe"

# Cheap STT for the stub bot. Bot replies are 60-token throwaways —
# accuracy matters less than cost.
OPENAI_BOT_STT_MODEL: str = "gpt-4o-mini-transcribe"

# Gemini Live — bidirectional audio. `latest` follows whatever Google
# promotes so this doesn't bitrot when previews shift.
GEMINI_LIVE_MODEL: str = "gemini-2.5-flash-native-audio-latest"

# OpenAI multimodal audio chat — used by the example openai_voice_agent
# helper (chat completions with audio modality, separate API surface
# from Realtime). `gpt-audio-mini` is the current cost-efficient GA
# tier; `gpt-4o-audio-preview` was the legacy preview model.
OPENAI_AUDIO_CHAT_MODEL: str = "gpt-audio-mini"

# Composable voice agent LLM — text-only, drives the brain of the
# `ComposableVoiceAgent` and `ElevenLabsVoiceAgent` stacks (STT → LLM
# → TTS). Litellm-style identifier ("openai/...", "anthropic/...").
COMPOSABLE_VOICE_LLM_MODEL: str = "openai/gpt-5.4-mini"
